components/builder-api-client/src/builder.rs

```diff
@@ -828,6 +829,10 @@ impl BuilderAPIProvider for BuilderAPIClient {
                .append_pair("checksum", &checksum)
                .append_pair("target", &target.to_string())
                .append_pair("forced", &force_upload.to_string());
+            
+            if let Some(val) = disable_build_group {
+              url.query_pairs_mut().append_pair("builder", val);
+            }
```

Comment:
Just an interesting rust FYI that `Option` and `Result` implement `IntoIterator`, which would allow you to do this as:
```rust
            url.query_pairs_mut()
               .extend_pairs(disable_build_group.map(|v| ("builder", v)));
```
Per my other comments, I think we should use a non-`Option` type here, but I thought this was handy to know.

Reply:
Just to be clear, if there's a `builder=XXX` item in the query parameters, then that **dis**ables the auto-build functionality? That's really unexpected. Why is the builder API like that?

Reply:
When Builder does an upload it wants to disable the auto build (since it is doing those builds, and we don't want an infinite loop). The query param naming is historical.

Reply:
That makes more sense; thanks for the context.